raspadorlegislativo/spiders/__init__.py

```diff
@@ -1,23 +1,65 @@
 import os
+from collections import namedtuple
+from functools import partial
 from tempfile import mkstemp
+from uuid import uuid4
 
-from scrapy import Spider
+from scrapy import Spider as OriginalSpider
+from memcache import Client
 
 from raspadorlegislativo import settings
 from raspadorlegislativo.utils.pdf import extract_text
 
 
-class Spider(Spider):
+PendingRequest = namedtuple('PendingRequest', ('request', 'url', 'callback'))
 
-    def parse_pdf(self, data, response):
+
+class Spider(OriginalSpider):
+
+    def __init__(self):
+        self.cache = Client((settings.MEMCACHED_LOCATION,))
+
+    def get_unique_id(self):
+        return str(uuid4()).encode()
+
+    def set_bill(self, uuid, data):
+        return self.cache.set(uuid, data, 60 * 60 * 3)
```

Comment:
60 * 60 * 3 stands for?

Reply:
60 seconds multiplied 60 times becomes 1h.
Multiplied by 3 becomes 3 hours.
3 hours is a little bit more than what it takes to run the bot.
So I'm implying the cache lives roughly while the bot runs in a dirty guess.